Add user facing doc for management-components upgrade (#7661)
aws/eks-anywhere: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md

---
title: "Upgrade management components"
linkTitle: "Upgrade management components"
weight: 21
aliases:
    /docs/tasks/cluster/cluster-upgrades/management-components-upgrade/
date: 2024-02-21
description: >
  How to perform management components upgrade for management clusters
---

> >**_NOTE:_** `upgrade management-components` sub-command is added in `eksctl anywhere` version `v0.19.0` for all providers. Management components upgrade can only be done through `eksctl-anywhere` cli, not through Kubernetes API.

Reviewer: ```suggestion
>**_NOTE:_** The `upgrade management-components` sub-command is added in `eksctl anywhere` version `v0.19.0` for all providers. Management components upgrades can only be done through the `eksctl anywhere` cli, not through the Kubernetes API.
```

### What are “management components”?

> Management components are the k8s resources in an EKSA management cluster that are responsible for managing the EKS-A clusters' lifecycle. Key components include, but are not limited to:

Reviewer: ```suggestion
Management components are the Kubernetes resources in an EKS Anywhere management cluster that are responsible for managing the EKS Anywhere cluster's lifecycle. Key components include, but are not limited to:
```

* The `eksa-controller-manager` pod located in the `eksa-system` namespace.

> * Tinkerbell services located in a bare metal management cluster.

Reviewer: ```suggestion
* Tinkerbell services located in a Bare Metal management cluster.
```

* Custom Resource Definitions (CRDs) such as `clusters`, `eksareleases`, and `tinkerbellmachineconfigs`.

> ### Why upgrade management-components separately?

Reviewer: ```suggestion
### Why upgrade management components separately?
```

> The “upgrade cluster” command, when executed against a management cluster, not only upgrades its management components but also initiates a comprehensive upgrade of the management cluster itself. This process often entails node replacement and significant operational tasks, which might be deemed risky for some production environments due to the potential of replacing all Kubernetes nodes. 

Reviewer: ```suggestion
The `upgrade cluster` command, when executed against a management cluster, not only upgrades its management components but also initiates a comprehensive upgrade of the management cluster itself. This process often entails node replacement and significant operational tasks, which might be deemed risky for some production environments due to the potential of replacing all Kubernetes nodes. 
```

> However, to leverage enhancements of newer EKSA version, such as bug fixes in the new management components without the extensive risk, it's advisable to initially upgrade only the management components. This step ensures minimal disruption and risk. Subsequently, you can proceed to upgrade the workload cluster at a more convenient time, once you're confident in the stability and readiness of your environment. This strategy provides a more controlled and efficient method to update your cluster's management infrastructure, aligning upgrades with your readiness and risk tolerance levels.

Reviewer: ```suggestion
However, to leverage enhancements to newer EKS Anywhere versions, such as bug fixes in the new management components without the extensive risk, it's advisable to initially upgrade only the management components. This step ensures minimal disruption and risk. Subsequently, you can proceed to upgrade the workload cluster at a more convenient time, once you're confident in the stability and readiness of your environment. This strategy provides a more controlled and efficient method to update your cluster's management infrastructure, aligning upgrades with your readiness and risk tolerance levels.
```

---
title: "Upgrade management components"
linkTitle: "Upgrade management components"
weight: 21
aliases:
    /docs/tasks/cluster/cluster-upgrades/management-components-upgrade/
date: 2024-02-21
description: >

>   How to perform management components upgrade for management clusters

Reviewer 2: How to upgrade EKS Anywhere management components

---

> >**_NOTE:_** The `upgrade management-components` sub-command is added in `EKS Anywhere` version `v0.19.0` for all providers. Management components upgrades can only be done through the `eksctl anywhere` cli, not through the Kubernetes API.

Reviewer 2: The `eksctl anywhere upgrade management-components` subcommand was added in EKS Anywhere version `v0.19.0` for all providers. Management component upgrades can only be done through the `eksctl` CLI, not through the Kubernetes API.

```

The output should appear similar to the following:

```
NAME                 CURRENT VERSION       NEXT VERSION
EKS-A Management     v0.18.3+cc70180       v0.19.0+a672f31
cert-manager         v1.13.0+68bec33       v1.13.2+a34c207
cluster-api          v1.5.2+b14378d        v1.6.0+04c07bc
kubeadm              v1.5.2+5762149        v1.6.0+5bf0931
vsphere              v1.7.4+6ecf386        v1.8.5+650acfa
etcdadm-bootstrap    v1.0.10+c9a5a8a       v1.0.10+1ceb898
etcdadm-controller   v1.0.16+0ed68e6       v1.0.17+5e33062
```

Alternatively, you can run the `eksctl anywhere upgrade plan cluster` command against your management cluster, which shows the version differences for both management and cluster components.

### Upgrade management components

To perform the management components upgrade, run the following command:

```
eksctl anywhere upgrade management-components -f management-cluster.yaml
```

The output should appear similar to the following:

```
Performing setup and validations
✅ Docker provider validation
✅ Control plane ready
✅ Cluster CRDs ready
Upgrading core components
Installing new eksa components
🎉 Management components upgraded!
```

> At this point, a new `eksaVersion` custom resource will be available in your management cluster. You can subsequently run a workload cluster upgrade with the `eksctl anywhere upgrade cluster command`, or by updating `eksaVersion` field in your workload cluster's spec and applying it to your management cluster with Kubernetes API-compatible tooling such as kubectl, GitOps, or Terraform.

Reviewer 2: At this point, a new `eksaVersion` custom resource will be available in your management cluster, which means new cluster components that correspond to the `eksaVersion` are available for cluster upgrades.
